daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionCoder.scala

```diff
@@ -354,7 +354,8 @@ object TransactionCoder {
           else if (txVersion precedes minContractKeyInFetch)
             Left(DecodeError(s"$txVersion is too old to support NodeFetch's `key` field"))
           else decodeKeyWithMaintainers(decodeCid, protoFetch.getKeyWithMaintainers).map(Some(_))
-        } yield (ni, NodeFetch(c, templateId, None, actingParties, signatories, stakeholders, key))
+        } yield
+          (ni, NodeFetch(c, templateId, None, actingParties, signatories, stakeholders, key, false))
```

Comment:
Why do we always set this to `false` during decoding? Does the encoded version not contain the required information? Same question a bit further down.

Reply:
The encoded version does not contain the required information.
We have to add the flag to `transaction.proto`. I will do that in a following up PR.  
This PR is concerned with replacing the field `PartialTransaction#byKeyNodes` by the `byKey` flag in `GenNode`.

Reply:
Thanks for the explanation.